From 7e4f74f5040fc3f4e7646695b54c5c11b0b95c62 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Mon, 11 Nov 2013 16:57:32 +0100
Subject: [PATCH] systemd: do not exit the initqueue, if systemd asks a
 password

this prevents bailing out the initqueue, while passwords are still to be
asked
---
 modules.d/98systemd/dracut-initqueue.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/modules.d/98systemd/dracut-initqueue.sh b/modules.d/98systemd/dracut-initqueue.sh
index d7ebf2b..64e8154 100755
--- a/modules.d/98systemd/dracut-initqueue.sh
+++ b/modules.d/98systemd/dracut-initqueue.sh
@@ -52,6 +52,10 @@ while :; do
     # no more udev jobs and queues empty.
     sleep 0.5
 
+    for i in /run/systemd/ask-password/ask.*; do
+        [ -e "$i" ] && continue
+    done
+
     if [ $main_loop -gt $((2*$RDRETRY/3)) ]; then
         for job in $hookdir/initqueue/timeout/*.sh; do
             [ -e "$job" ] || break
